### PR TITLE
Issue #881: Issue #809 still occurs for reverse scans on forward cfs

### PR DESCRIPTION
**Pull request by spetrunia**
When constructing a lookup key for reverse full index scan, pass the
correct eq_cond_len to setup_scan_iterator(). The code mirrors the
code in ha_rocksdb::index_first_intern().

### CODE REVIEW

File: storage/rocksdb/rdb_datadic.h [hunk 1]
```diff
@@ -256,6 +256,27 @@ class Rdb_key_def {
...
+  /*
+    The same as get_first_key, but get the key for the last entry in the index
+  */
+  inline int get_last_key(uchar *const key, uint *const size) const {
+    if (m_is_reverse_cf)
+      get_infimum_key(key, size);
+    else
+      get_supremum_key(key, size);
+
+    /* Find out how many bytes of infimum are the same as m_index_number */
```
Comment (lth) [inline storage/rocksdb/rdb_datadic.h:268]: Update the comment?

File: storage/rocksdb/rdb_datadic.h [hunk 2]
```diff
@@ -256,6 +256,27 @@ class Rdb_key_def {
...
+    rdb_netbuf_store_index(unmodified_key, m_index_number);
+    int i;
+    for (i = 0; i < INDEX_NUMBER_SIZE; i++) {
+      if (key[i] != unmodified_key[i])
+        break;
+    }
+    return i;
+  }
+
+
```
Comment (lth) [inline storage/rocksdb/rdb_datadic.h:279]: clang-format is complaining about new line here.

**Review by lth — APPROVED** [review]
Changes look good to me.

It would be good to fix comments though.

**Review by hermanlee — APPROVED** [review]

File: storage/rocksdb/rdb_datadic.h (hunk 2 above)
Comment (hermanlee) [reply storage/rocksdb/rdb_datadic.h:279]: I'll fix these separately.

**Pull request closed without merging**